src/main/java/wolox/training/services/UserService.java

```diff
@@ -0,0 +1,29 @@
+package wolox.training.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@Service
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    public User persistUser(User user) {
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
```

Comment:
Nice! But you could avoid this by putting the encoder in the setter of the password directly.